### PR TITLE
config, tso: add labels configuration to update TSO configuration

### LINKED ISSUE #3414: Change the configuration way of Local TSO
### Describe your feature request related problem

For now, we use `local-tso.dc-location` and `local-tso.enable-local-tso` to configure the Local TSO, which is kind of inconvenient and inconsistent with TiDB and TiKV.

### Describe the feature you'd like

We should introduce the feature of `labels` like TiDB and TiKV, which will be more uniform and reasonable for users to use.

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Close #3414.

### What is changed and how it works?

* Introduce `Labels` configuration.
* Make TSO use labels to configure.

### Check List

Tests

- Unit test
- Integration test

Related changes

- pingcap/tidb#19212
- pingcap/tidb#22639

### CODE REVIEW

File: conf/config.toml [hunk 1]
```diff
@@ -20,8 +20,17 @@ initial-cluster-state = "new"
...
 
+## Make sure you set the "zone" label for this PD server before enabling its Local TSO service.
+# enable-local-tso = true
+
 enable-prevote = true
 
+[labels]
+foo= "bar"
+group= "abc"
+# This means **this** PD server belongs to data center "dc-1".
```
Comment (Yisaer) [inline conf/config.toml:31]: I think we can remove these 2 labels.

**Review by Yisaer — COMMENTED** [review]
Rest LGTM

**Review by Yisaer — APPROVED** [review]
LGTM

File: conf/config.toml [hunk 2]
```diff
@@ -20,8 +20,15 @@ initial-cluster-state = "new"
...
 tso-save-interval = "3s"
 
+## Make sure you set the "zone" label for this PD server before enabling its Local TSO service.
+# enable-local-tso = true
+
 enable-prevote = true
 
+[labels]
+# This means **this** PD server belongs to data center "dc-1".
+zone="dc-1"
```
Comment (HunDunDM) [inline conf/config.toml:30]: Need to comment out this section?

File: server/config/config.go [hunk 3]
```diff
@@ -93,8 +93,11 @@ type Config struct {
 	// be automatically clamped to the range.
 	TSOUpdatePhysicalInterval typeutil.Duration `toml:"tso-update-physical-interval" json:"tso-update-physical-interval"`
 
-	// Local TSO service related configuration.
-	LocalTSO LocalTSOConfig `toml:"local-tso" json:"local-tso"`
+	// EnableLocalTSO is used to enable the Local TSO Allocator feature,
+	// which allows the PD server to generate Local TSO for certain DC-level transactions.
+	// To make this feature meaningful, user has to set the "zone" label for the PD server
+	// to indicate which DC this PD belongs to.
+	EnableLocalTSO bool `toml:"enable-local-tso" json:"enable-local-tso"`
```
Comment (HunDunDM) [inline server/config/config.go:100]: Incompatible changes need to be confirmed whether it has no effect on downstream.

File: conf/config.toml (hunk 2 above)
Comment (JmPotato) [reply conf/config.toml:30]: No need to comment out because there are no side effects I think.

File: server/config/config.go (hunk 3 above)
Comment (JmPotato) [reply server/config/config.go:100]: This feature is not released yet and there is no effect on downstream. It's ok to change.

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**